backend/sublet/views.py

```diff
@@ -0,0 +1,213 @@
+
+
+class Properties(viewsets.ModelViewSet):
+    """
+    list:
+    Returns a list of Sublets that match query parameters (e.g., amenities) and belong to the user.
+
+    create:
+    Create a Sublet.
+
+    partial_update:
+    Update certain fields in the Sublet. Only the owner can edit it.
+
+    destroy:
+    Delete a Sublet.
+    """
+
+    # how to use the sublet owner permission
+    permission_classes = [SubletOwnerPermission | IsSuperUser]
+    serializer_class = SubletSerializer
+
+    def get_queryset(self):
+        # All Sublets for superusers
+        # if self.request.user.is_superuser:
+        #     return Sublet.objects.all()
+
+        # # All Sublets where expires_at hasn't passed yet for regular users
+        # return Sublet.objects.filter(expires_at__gte=timezone.now())
+        return Sublet.objects.all()
+
+    # def create(self, request, *args, **kwargs):
+    #     # amenities = request.data.pop("amenities", [])
+    #     new_data = request.data
+    #     amenities = new_data.pop("amenities", [])
+
+    #     # check if valid amenities
+    #     try:
+    #         amenities = [Amenity.objects.get(name=amenity) for amenity in amenities]
+    #     except Amenity.DoesNotExist:
+    #         return Response({"amenities": "Invalid amenity"}, status=status.HTTP_400_BAD_REQUEST)
+
+    #     serializer = self.get_serializer(data=new_data)
+    #     serializer.is_valid(raise_exception=True)
+    #     sublet = serializer.save()
+    #     sublet.amenities.set(amenities)
+    #     sublet.save()
+    # return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def list(self, request, *args, **kwargs):
+        """Returns a list of Sublets that match query parameters and user ownership."""
+        # Get query parameters from request (e.g., amenities, user_owned)
+        amenities = request.query_params.getlist("amenities")
+        subletter = request.query_params.get(
+            "subletter", "false"
+        )  # Defaults to False if not specified
+        starts_before = request.query_params.get("starts_before", None)
+        starts_after = request.query_params.get("starts_after", None)
+        ends_before = request.query_params.get("ends_before", None)
+        ends_after = request.query_params.get("ends_after", None)
+        min_price = request.query_params.get("min_price", None)
+        max_price = request.query_params.get("max_price", None)
+        beds = request.query_params.get("beds", None)
+        baths = request.query_params.get("baths", None)
+
+        queryset = Sublet.objects.all().filter(expires_at__gte=timezone.now())
+
+        # Apply filters based on query parameters
+        if amenities:
+            queryset = queryset.filter(amenities__name__in=amenities)
+        if subletter.lower() == "true":
+            queryset = queryset.filter(subletter=request.user)
+        if starts_before:
+            queryset = queryset.filter(start_date__lt=starts_before)
```

Comment:
nit: can we test one of two of these if statements just to guarantee that they work? no need to test them all unless you want to